### PR TITLE
feat(signer): Eth tests with papi

### PR DESCRIPTION
**Pull request by bitdivine**
# Motivation
Re-enable all Ethereum tests, this time with payment.

# Changes
- Move all ethereum tests from `sign.rs` and `address.rs` to `eth.rs`.
- Update the tests to provide payment.

# Tests
These tests are all exercised in CI

### CODE REVIEW

File: src/signer/canister/tests/it/sign.rs
Comment (bitdivine) [inline src/signer/canister/tests/it/sign.rs:1]: I actually renamed this file to eth.rs, as all the tests in here were for ethereum endpoints.  But I guess adding payment made such a large delta that git is treating it as a  new file.

File: src/signer/canister/tests/it/address.rs
Comment (bitdivine) [inline src/signer/canister/tests/it/address.rs:1]: I moved all the ethereum address tests into `eth.rs` as I think they fit better there.

**Comment by bitdivine** [comment]
Demo:
```mermaid
sequenceDiagram
  actor User
  participant AppFrontend
  participant AppBackend
  participant cfSigner
  participant cyclesLedger
  User ->> AppFrontend: Request Action
  AppFrontend ->> AppBackend: Notify
  AppBackend ->> cyclesLedger: Optional: transfer(somecycles, to: userAppSubaccount)
  cyclesLedger -->> AppBackend: 
  AppBackend -->> AppFrontend: Notify OK
  AppFrontend ->> cyclesLedger: icrc2_approve(somecycles, to: cfSignerDefaultAccount)
  cyclesLedger -->> AppFrontend: 
  AppFrontend->>cfSigner: please_sign(tosign)
  cfSigner->>cyclesLedger: icrc2_transfer_from(user, somecycles)
  cyclesLedger-->>cfSigner: Balance transferred
  cfSigner-->>AppFrontend: Signed Message
  AppFrontend-->>Ethereum: Signed Message
  AppFrontend-->>User: Action Complete
```

**Review by lmuntaner — APPROVED** [review]
LGTM! Thanks

**Pull request merged**